pysilcam/acquisition.py

```diff
@@ -138,122 +284,222 @@ def get_generator_disc(self, datapath=None, writeToDisk=False, camera_config_fil
+            # Prepare for image acquisition and create a frame
             frame0 = camera.getFrame()
             frame0.announceFrame()
 
-            #Aquire raw images and yield to calling context
+            # Aquire raw images and yield to calling context
             while True:
                 try:
                     timestamp, img = self._acquire_frame(camera, frame0)
                     yield timestamp, img
                 except Exception:
                     frame0.img_idx += 1
                     if frame0.img_idx > len(frame0.files):
-                        #print('  END OF FILE LIST.')
                         logger.info('  END OF FILE LIST.')
                         break
 
+    def image_handler(self, camera, frame):
+        logger.debug('camera image handler')
+        with camera:
+            if frame.get_status() == FrameStatus.Complete:
 
-    def get_generator_camera(self, datapath=None, writeToDisk=False, camera_config_file=None):
-        '''
-        Aquire images from Silcam
-        
-        Args:
-            datapath: path from where the images are acquired.
-            writeToDisk: boolean indicating wether the acquired 
-                images have to be written to disc.
+                img = frame.as_numpy_ndarray()
 
-        Yields:
-            timestamp: timestamp of the acquired image.
-            img: acquired image.
+                timestamp = pd.Timestamp.now()
+                filename = os.path.join(self.datapath, timestamp.strftime('D%Y%m%dT%H%M%S.%f.silc'))
+
+                if self.writeToDisk:
+                    with open(filename, 'wb') as fh:
+                        np.save(fh, img, allow_pickle = False)
+                        fh.flush()
+                        os.fsync(fh.fileno())
+
+                # previously we calculated acquisition frequency here
+
+                # gui data handling
+                self.gui_update(timestamp, img)
+
+                if self.raw_image_queue is not None:
+                    # Prev put_nowait() !!!!!
+                    logger.debug(f"timestamp+img adding to stack: {datetime.now()}")
+                    try:
+                        self.raw_image_queue.put_nowait((-1, timestamp, img))
+                    except:
+                        pass
+                    # self.raw_image_queue.put((timestamp, img))
+                    logger.debug(f"timestamp+img added to stack:  {datetime.now()}")
+
+            camera.queue_frame(frame)  # ask the camera for the next frame, which would evtentually call image_handler again
+
+    def gui_update(self, timestamp, img):
+        if self.gui is not None:
+            while (self.gui.qsize() > 0):
+                try:
+                    self.gui.get_nowait()
+                    time.sleep(0.001)
+                except:
+                    continue
+
+            rtdict = dict()
+            rtdict = {'dias': 0,
+                        'vd_oil': 0,
```

Comment:
E127: continuation line over-indented for visual indent  <!-- Automatically posted by Lintly -->